File: app/src/main/java/com/example/syl/android_design_patterns/MainActivity.java
package com.example.syl.android_design_patterns;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.syl.android_design_patterns.builder_example.User;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        User user = new User.UserBuilder("Pitt", "Smith").age(24).phone("+34000000000").address("Address").build();

        System.out.println(user.toString());
    }
}

File: app/src/main/java/com/example/syl/AndroidDesignPatterns/MainActivity.kt
package com.example.syl.AndroidDesignPatterns

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.example.syl.AndroidDesignPatterns.decorator_example.java.BaseProduct
import com.example.syl.AndroidDesignPatterns.decorator_example.java.decorations.ProductImage
import com.example.syl.AndroidDesignPatterns.decorator_example.java.decorations.ProductTitle

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val productImageTitle = ProductTitle(ProductImage(BaseProduct()))
        productImageTitle.makeProduct()
    }
}